lib/resource_waiter.js

```diff
@@ -136,6 +136,16 @@ AWS.ResourceWaiter = inherit({
       callback = params; params = undefined;
     }
 
+    if (params && params.$waiter) {
+      if (params.$waiter.delay) {
+        this.config.delay = params.$waiter.delay;
+      }
+      if (params.$waiter.maxAttempts) {
+        this.config.maxAttempts = params.$waiter.maxAttempts;
+      }
+      delete params.$waiter;
```

Comment:
Good point. I added a call to AWS.util.copy to clone the params so that we're not passing any unknown keys to the service operation.